provisioning/miniprov/hare_mp/main.py

```diff
@@ -83,31 +83,49 @@ def _report_unsupported_features(features_unavailable):
         uf_db.store_unsupported_features('hare', features_unavailable))
 
 
-def logrotate():
+def logrotate(url: str):
     try:
-        setup_info = get_data_from_provisioner_cli('get_setup_info')
-        if setup_info != 'unknown':
-            server_type = setup_info['server_type']
-            shutil.copyfile(
-                f'/opt/seagate/cortx/hare/conf/logrotate/{server_type}',
-                '/etc/logrotate.d/hare')
+        provider = ConfStoreProvider(url)
+        machine_id = provider.get_machine_id()
+        server_type = provider.get(
+            f'server_node>{machine_id}>type')
```

Comment:
Let's avoid the code duplication, please move the code in a function, e.g. get_setup_info(url: str) and invoke it from the main(), unconditionally and pass the type to both the functions (logrotate() and unsupported_feature()).

Reply:
Done